test/integration/apimachinery/check_list_fields_test.go

```diff
@@ -0,0 +1,104 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package apimachinery
+
+import (
+	"context"
+	"k8s.io/apimachinery/pkg/fields"
+	"testing"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/client-go/kubernetes"
+	kubeapiservertesting "k8s.io/kubernetes/cmd/kube-apiserver/app/testing"
+	"k8s.io/kubernetes/test/integration/framework"
+)
+
+func TestCheckFieldItemsInEmptyList(t *testing.T) {
+	server := kubeapiservertesting.StartTestServerOrDie(t, nil, []string{
+		"--runtime-config=api/all=true",
+	}, framework.SharedEtcd())
+
+	defer server.TearDownFn()
+	server.ClientConfig.AcceptContentTypes = "application/json"
+	server.ClientConfig.ContentType = "application/json"
+
+	clientSet, err := kubernetes.NewForConfig(server.ClientConfig)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	_, lists, err := clientSet.Discovery().ServerGroupsAndResources()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	for _, resources := range lists {
+		for _, resource := range resources.APIResources {
+			if !sets.NewString(resource.Verbs...).Has("list") {
+				t.Logf("skip gv: %s, kind: %s", resources.GroupVersion, resource.Kind)
+				continue
+			}
+
+			var prefix []string
+			if resources.GroupVersion == "v1" {
+				prefix = []string{"api", "v1"}
+			} else {
+				prefix = []string{"apis", resources.GroupVersion}
+			}
+
+			request := clientSet.RESTClient().Get().AbsPath(prefix...).
+				NamespaceIfScoped("non-exists-namespace", resource.Namespaced).
+				Resource(resource.Name).
+				VersionedParams(&metav1.ListOptions{
```

Comment:
I think this can be significantly simplified using the dynamic client to make list requests instead of RESTClient() + manually constructing the URL

```go
dynamicClient, err := dynamic.NewForConfig(config)
if err != nil {
  t.Fatal(...)
}
list, err := dynamicClientSet.Resource(groupVersionResource).List(context.Background(), metav1.ListOptions{LabelSelector: "non-exist=non-exist"})
if err != nil {
  t.Fatal(...)
}
if list.Items == nil {
  t.Fatal(...)
}
if len(list.Items) > 0 {
  t.Fatal(...)
}
```